build/build.go

```diff
@@ -703,11 +706,49 @@ func Invoke(ctx context.Context, cfg ContainerConfig) error {
 		}
 		defer ctr.Release(context.TODO())
 
+		imgData := res.Metadata[exptypes.ExporterImageConfigKey]
+		var img *specs.Image
+		if len(imgData) > 0 {
+			img = &specs.Image{}
+			if err := json.Unmarshal(imgData, img); err != nil {
+				fmt.Println(err)
+				return nil, err
+			}
+		}
+
+		user := ""
+		if cfg.User != nil {
+			user = *cfg.User
+		} else if img != nil {
+			user = img.Config.User
+		}
+		cwd := ""
+		if cfg.Cwd != nil {
+			cwd = *cfg.Cwd
+		} else if img != nil {
+			cwd = img.Config.WorkingDir
+		}
+		env := cfg.Env
+		if env == nil && img != nil {
+			env = img.Config.Env
```

Comment:
I think image env should be always set. If user sets own env they are added on top of image like in `docker run`